### PR TITLE
chore: organize and document top level dev scripts

### PR DESCRIPTION
**Pull request by hwkr**
Organizes and documents all the top level turbo dev scripts, let me know if this doesn't make sense.

Google Chrome-2024-10-16-13-38-33@2x

### CODE REVIEW

**Comment by hwkr** [comment]
> Funny I was just looking for the same thing. looks like its called dev:web-app now. I would rather call it client though. Maybe delete that one and just keep the one you changed
> 
> "dev:web-app": "turbo playground:web --filter @scalar/api-client",

The playground doesn't have a watcher or HMR as far as I can tell

**Review by amritk — APPROVED** [review]
cleen

**Review by hanspagel — APPROVED** [review]
So fresh so clean

File: README.md [hunk 1]
```diff
@@ -181,7 +181,7 @@ We are API nerds. You too? Let’s chat on Discord: <https://discord.gg/scalar>
 
 ### Contributions
 
-Contributions are welcome! Read the [`CONTRIBUTING`](https://github.com/scalar/scalar/blob/main/CONTRIBUTING) guide.
+Contributions are welcome! Read the [`CONTRIBUTING`](https://github.com/scalar/scalar/blob/main/CONTRIBUTING.md) guide.
```
Comment (hanspagel) [inline README.md:184]: nope

**Review by hanspagel — CHANGES_REQUESTED** [review]

**Review by hanspagel — APPROVED** [review]

**Pull request merged**